stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java

```diff
@@ -94,14 +112,96 @@ public static MimeMessage generateMessage(Session session, String username, MapV
             emailMessage.setReplyTo(replyToAddressArray);
         }
         emailMessage.setSubject(subject);
-        emailMessage.setText(messageBody);
         emailMessage.setFrom(new InternetAddress(fromAddress));
         if (!senderAddress.isEmpty()) {
             emailMessage.setSender(new InternetAddress(senderAddress));
         }
+        ArrayValue attachments = message.getArrayValue(EmailConstants.MESSAGE_ATTACHMENTS);
+        if (attachments == null) {
+            emailMessage.setText(messageBody);
+        } else {
+            addBodyAndAttachments(emailMessage, messageBody, attachments);
+        }
         return emailMessage;
     }
 
+    private static void addBodyAndAttachments(MimeMessage emailMessage, String messageBody, ArrayValue attachments)
+            throws MessagingException, IOException {
+        BodyPart messageBodyPart = new MimeBodyPart();
+        messageBodyPart.setText(messageBody);
+        Multipart multipart = new MimeMultipart();
+        multipart.addBodyPart(messageBodyPart);
+        for (int i = 0; i < attachments.size(); i++) {
+            if (attachments.get(i) instanceof ObjectValue) {
```

Comment:
What happens if this condition does not satisfied?

Reply:
It will be skipped getting added as an attachment. Anyway, this case will not happen according to the referred code of this method.

Reply:
Then it's better to add it as a comment at least I guess. This may lead to confusions in the future.

Reply:
I think that is implicit as there can be emails without attachments too.